lib/vrl/compiler/src/expression/function_call.rs

```diff
@@ -360,6 +360,23 @@ impl<'a> Builder<'a> {
 
                                     (kind.into(), None)
                                 }
+
+                                // The variable kind is expected to be equal to
+                                // the kind of all keys within the target
+                                // collection type.
+                                //
+                                // This means it's either a collection of 
+                                // strings for an object or a collection of
+                                // integers for an array.
+                                VariableKind::TargetInnerKeys => {
+                                    let mut kind = Kind::never();
+
+                                    if type_def.is_array() {
+                                        kind.add_array(value::kind::Collection::any());
+                                    } 
+             
+                                    (kind.into(), None)
+                                }
```

Comment:
As a rust newbie I was unsure on the correct implementation here - before I spend too long looking at this would you be able to suggest the appropriate way for arrays of bytes/ints 😬 ?

Reply:
Other than that and type definitions that'll need updating - these changes have made it that's a bit closer

Reply:
`kind.add_array(Collection::from_unknown(Kind::integer()))` adds an array of integers. The "unknown" part just means the exact length isn't known, so it's an array of unknown length containing integers.

Similarly for an array of bytes/strings
`kind.add_array(Collection::from_unknown(Kind::bytes()))`